src/desktops/dskGameInterface.cpp

```diff
@@ -1017,6 +1020,75 @@ void dskGameInterface::OnChatCommand(const std::string& cmd)
             worldViewer.InitTerrainRenderer();
         }
     }
+    else if(cmd == "ironore") cheatAdd(GD_IRONORE, 10, cmd);
+    else if(cmd == "beer") cheatAdd(GD_BEER, 10, cmd);
+    else if(cmd == "tongs") cheatAdd(GD_TONGS, 10, cmd);
+    else if(cmd == "hammer") cheatAdd(GD_HAMMER, 10, cmd);
+    else if(cmd == "axe") cheatAdd(GD_AXE, 10, cmd);
+    else if(cmd == "saw") cheatAdd(GD_SAW, 10, cmd);
+    else if(cmd == "pickaxe") cheatAdd(GD_PICKAXE, 10, cmd);
+    else if(cmd == "shovel") cheatAdd(GD_SHOVEL, 10, cmd);
+    else if(cmd == "crucible") cheatAdd(GD_CRUCIBLE, 10, cmd);
+    else if(cmd == "rodandline") cheatAdd(GD_RODANDLINE, 10, cmd);
+    else if(cmd == "scythe") cheatAdd(GD_SCYTHE, 10, cmd);
+    else if(cmd == "cleaver") cheatAdd(GD_CLEAVER, 10, cmd);
+    else if(cmd == "rollingpin") cheatAdd(GD_ROLLINGPIN, 10, cmd);
+    else if(cmd == "bow") cheatAdd(GD_BOW, 10, cmd);
+    else if(cmd == "boat") cheatAdd(GD_BOAT, 10, cmd);
+    else if(cmd == "iron") cheatAdd(GD_IRON, 10, cmd);
+    else if(cmd == "flour") cheatAdd(GD_FLOUR, 10, cmd);
+    else if(cmd == "fish") cheatAdd(GD_FISH, 10, cmd);
+    else if(cmd == "bread") cheatAdd(GD_BREAD, 10, cmd);
+    else if(cmd == "wood") cheatAdd(GD_WOOD, 10, cmd);
+    else if(cmd == "boards") cheatAdd(GD_BOARDS, 10, cmd);
+    else if(cmd == "stones") cheatAdd(GD_STONES, 10, cmd);
+    else if(cmd == "grain") cheatAdd(GD_GRAIN, 10, cmd);
+    else if(cmd == "coins") cheatAdd(GD_COINS, 10, cmd);
+    else if(cmd == "gold") cheatAdd(GD_GOLD, 10, cmd);
+    else if(cmd == "coal") cheatAdd(GD_COAL, 10, cmd);
+    else if(cmd == "meat") cheatAdd(GD_MEAT, 10, cmd);
+    else if(cmd == "ham") cheatAdd(GD_HAM, 10, cmd);
+    else if(cmd == "water") cheatAdd(GD_WATER, 10, cmd);
+    else if(cmd == "givemeall") cheatAddAll();
+}
+
+void dskGameInterface::cheatAdd(GoodType type, unsigned value, const std::string& type_string) {
+	if(isCheatModeOn) {
+		GameWorldBase& world = worldViewer.GetWorldNonConst();
+		GamePlayer& player = world.GetPlayer(worldViewer.GetPlayerId());
+
+		nobBaseWarehouse* wh = player.GetFirstWH();
+		Inventory goods;
+		goods.Add(type, value);
+		wh->AddGoods(goods, true);
+
+		if(type_string.length() != 0) {
+			char text[256];
+			snprintf(text, sizeof(text), _("Cheat: Added %d %s"), value, type_string.c_str());
+			messenger.AddMessage("", 0, CD_SYSTEM, text, COLOR_RED);
+		}
+	}
+}
+
+void dskGameInterface::cheatAddAll(unsigned value=10){
+	if(isCheatModeOn) {
+		GoodType arr[] = { GD_IRONORE, GD_BEER, GD_TONGS, GD_HAMMER, GD_AXE, GD_SAW,
```

Comment:
Use a `boost::array` with a fixed size. This also saves the copy to a list below.